source/funkin/ui/freeplay/FreeplayState.hx

```diff
@@ -638,6 +640,16 @@ class FreeplayState extends MusicBeatSubState
       onDJIntroDone();
     }
 
+    allDifficulties = [];
+    for (song in songs)
```

Comment:
I'd recommend moving this chunk of code to the SongRegistry and calling `allDifficulties = SongRegistry.instance.listAllDifficulties()`